elixir/apps/web/lib/web/live/resources/components.ex

```diff
@@ -57,119 +54,120 @@ defmodule Web.Resources.Components do
-              name={"#{@form.name}[icmp][enabled]"}
-              checked={Map.has_key?(@forms_by_protocol, :icmp)}
-              value="true"
-              disabled={Map.has_key?(@forms_by_protocol, :all)}
-              label="ICMP"
-            />
+      <p class="text-sm text-neutral-500">
+        Restrict access to the specified protocols and ports. By default, <strong>all</strong>
+        protocols and ports are accessible.
+      </p>
+
+      <div class={@traffic_filters_enabled? == false && "opacity-50"}>
+        <div>
+          <.input type="hidden" name={"#{@form.name}[icmp][protocol]"} value="icmp" />
+
+          <div class="items-center flex flex-row h-12">
+            <div class="flex-none w-32">
+              <.input
+                title="Allow ICMP traffic"
+                type="checkbox"
+                field={@forms_by_protocol[:icmp]}
+                name={"#{@form.name}[icmp][enabled]"}
+                checked={Map.has_key?(@forms_by_protocol, :icmp)}
+                value="true"
+                disabled={!@traffic_filters_enabled?}
+                label="ICMP"
+              />
+            </div>
           </div>
         </div>
-      </div>
 
-      <div>
-        <.input type="hidden" name={"#{@form.name}[tcp][protocol]"} value="tcp" />
-
-        <div class="items-center flex flex-row h-12">
-          <div class="flex-none w-32">
-            <.input
-              type="checkbox"
-              field={@forms_by_protocol[:tcp]}
-              name={"#{@form.name}[tcp][enabled]"}
-              checked={Map.has_key?(@forms_by_protocol, :tcp)}
-              value="true"
-              disabled={Map.has_key?(@forms_by_protocol, :all)}
-              label="TCP"
-            />
-          </div>
-
-          <div class="flex-grow">
-            <% ports = (@forms_by_protocol[:tcp] || %{ports: %{value: []}})[:ports] %>
-            <.input
-              type="text"
-              field={ports}
-              name={"#{@form.name}[tcp][ports]"}
-              value={pretty_print_ports(ports.value)}
-              disabled={Map.has_key?(@forms_by_protocol, :all)}
-              placeholder="Enter comma-separated port range(s), eg. 433, 80, 90-99. Matches all ports if empty."
-            />
+        <div>
+          <.input type="hidden" name={"#{@form.name}[tcp][protocol]"} value="tcp" />
+
+          <div class="items-center flex flex-row h-12">
+            <div class="flex-none w-32">
+              <.input
+                title="Restrict traffic to TCP traffic"
+                type="checkbox"
+                field={@forms_by_protocol[:tcp]}
+                name={"#{@form.name}[tcp][enabled]"}
+                checked={Map.has_key?(@forms_by_protocol, :tcp)}
+                value="true"
+                disabled={!@traffic_filters_enabled?}
+                label="TCP"
+              />
+            </div>
+
+            <div :if={Map.has_key?(@forms_by_protocol, :tcp)} class="flex-grow">
```

Comment:
I added this `:if` to only show the ports input when the port filter was enabled. Otherwise, it was possible to fill out the ports input and forget to check the box, which would prevent the traffic filter from being saved.

Reply:
I like the idea of not being able to fill out the ports input if the protocol is not selected, but rather than hide the input altogether, what do you think of just setting the input to `disabled` if the protocol isn't selected?

Reply:
Updated